pinot-spi/src/main/java/org/apache/pinot/spi/config/table/IndexingConfig.java

```diff
@@ -410,6 +411,14 @@ public void setSegmentNameGeneratorType(String segmentNameGeneratorType) {
     _segmentNameGeneratorType = segmentNameGeneratorType;
   }
 
+  public boolean isIndexCapacityThresholdCheckEnabled() {
+    return _indexCapacityThresholdCheckEnabled;
+  }
+
+  public void setIndexCapacityThresholdCheckEnabled(boolean indexCapacityThresholdCheckEnabled) {
+    _indexCapacityThresholdCheckEnabled = indexCapacityThresholdCheckEnabled;
+  }
+
```

Comment:
We may also want to set this for all tables. I would recommend (maybe in a future PR) to add an instance level configuration option to be the default value for this config. That way users can change the property at startup and that way they won't need to change hundreds of tables.

Reply:
> We may also want to set this for all tables
Did we encounter such scenario where there was need for this? In my knowledge, I am only aware of this issue happening only 1 time for 1 table only (rare).

Making this change as default can be dangerous (can create high num of segments in cluster) since we are taking a conservative approach over here.
There are indexes like FixedBitMVEntryDictForwardIndex, VarByteChunkForwardIndex, etc which are capable of holding more than 450 million num of entries and more than 2GB of data. 

For future PR, I think we should just work on making FixedBitMVForwardIndex unbounded and then we can enable this flag for default.

Reply:
Given the overhead is very low, I'd suggest making this on by default, or simply remove this flag.
We didn't run into this issue often because it is extremely rare to have more than 450M values in a segment. Before this PR, we usually manually reduce the commit threshold to workaround this, which is not ideal, and the user experience is very bad.